digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/AddressLinkBuilder.scala

```diff
@@ -201,8 +201,7 @@ trait AddressLinkBuilder {
       nextSegment.linkId == previousSegment.linkId &&
       nextSegment.geometry.nonEmpty && previousSegment.geometry.nonEmpty && // Check if geometries are not empty
       addressConnected(nextSegment, previousSegment) &&
-      !(cpNext._1.isDefined && cpPrevious._2.isDefined)) { // Check that the calibration point isn't between these segments
-
+      !((cpNext._1.isDefined || cpPrevious._2.isDefined && cpPrevious._2.get.addressMValue == nextSegment.startAddrMValue) && (cpPrevious._2.isDefined  || cpNext._1.isDefined && cpNext._1.get.addressMValue == previousSegment.endAddrMValue))){ // Check that the calibration point isn't between these segments
```

Comment:
This is not needed anymore. Because we have the calibration issue fixed on the convertion database.